cfme/tests/configure/test_analysis_profiles.py

```diff
@@ -333,3 +427,79 @@ def test_analysis_profile_description_validation(analysis_profile_collection):
+        # product_features tree for Managing Quotas for custom tenant
+        PRODUCT_FEATURES[1],
+        # product_features tree for Managing Dialogs for 'Add'
+        PRODUCT_FEATURES[2],
+        # product_features tree for Managing Dialogs for 'Edit'
+        PRODUCT_FEATURES[3],
+        # product_features tree for Managing Dialogs for 'Delete'
+        PRODUCT_FEATURES[4],
+        # product_features tree for Managing Dialogs for  'Copy'
+        PRODUCT_FEATURES[5]
+    ]
+)
+def test_custom_role_modify_for_dynamic_product_feature(request, appliance, product_features):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/12h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: Configuration
+        tags: quota
+        testSteps:
+            1. create two tenants
+            2. create new custom role using existing role
+            3. Update newly created custom role by doing uncheck in to options provided under
+               automation > automate > customization > Dialogs > modify > edit/add/copy/delete
+               > uncheck for any tenant
+            4. Or Update newly created custom role by doing uncheck in to options provided under
+               Settings > Configuration > Access Control > Tenants > Modify > Manage Quotas
+               > uncheck for any tenant
+            5. You will see save button is not enabled but if you changed 'Name' or
+               'Access Restriction for Services, VMs, and Templates' then save button is getting
+               enabled.
+            6. It updates changes only when we checked or unchecked for all of the tenants under
+               edit/add/copy/delete options.
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5. 'save' button should be enabled after changing product feature tree.
+            6. It should work for individual tenant.
+
+    Bugzilla:
+        1655012
+    """
+    tenant = appliance.collections.tenants.create(
+        name=TENANT_NAME,
+        description="tenant_des{}".format(fauxfactory.gen_alphanumeric()),
+        parent=appliance.collections.tenants.get_root_tenant(),
+    )
+    request.addfinalizer(tenant.delete)
+    role = appliance.collections.roles.instantiate(name='EvmRole-tenant_quota_administrator')
+    copied_role = role.copy(name="{role}_{name}".format(
+        role=role.name, name=fauxfactory.gen_alpha()))
+    request.addfinalizer(copied_role.delete)
+    view = navigate_to(copied_role, 'Details')
+
+    # Checks whether feature tree path is checked for given node
+    assert not view.features_tree.check_uncheck_node(True, *(product_features[0][0]))
+    copied_role.update({'product_features': product_features})
+
+    # Checks whether feature tree path is unchecked for given node
+    assert view.features_tree.check_uncheck_node(True, *(product_features[0][0]))
```

Comment:
I think you mean to pass `False` as the first argument to `check_uncheck_node()` here to uncheck?

Reply:
@mshriver 
- I think ```check_uncheck_node()``` is used to check or uncheck the node.
- So in this test case i need to check only; node is checked or not. Hence I am using ```node_checked```.